rust_src/src/lisp.rs

```diff
@@ -150,11 +151,37 @@ impl LispObject {
 }
 
 // Symbol support (LispType == Lisp_Symbol == 0)
-
 impl LispObject {
+    #[inline]
     pub fn is_symbol(self) -> bool {
         self.get_type() == LispType::Lisp_Symbol
     }
+
+    #[inline]
+    pub fn as_symbol(&self) -> Option<LispSymbolRef> {
+        if self.is_symbol() {
+            Some(LispSymbolRef::new(unsafe { mem::transmute(self.get_untaggedptr()) }))
```

Comment:
This looks to be incorrect. The code in lisp.h for XSYMBOL looks like
```
  intptr_t i = (intptr_t) XUNTAG (a, Lisp_Symbol);
  void *p = (char *) lispsym + i;
  return p;
```

and it looks like we will have to emulate this logic for getting the address to mem::transmute.